libsql-server/src/connection/libsql.rs

```diff
@@ -537,8 +545,20 @@ impl<W: WalHook> Connection<W> {
                 }
                 // lock was downgraded, notify a waiter
                 (Tx::Write, Tx::None | Tx::Read) => {
-                    state.slot.write().take();
-                    lock.slot.take();
+                    let old_slot = lock
```

Comment:
I wonder what are the cases here of a race? I don't know the full picture so hard to say but how can we ensure this is true?

Reply:
The slot could have been stolen after we had executed the statement but before we updated the slot in the state. In this case, the stealing txn would rollback our txn, but that's fine, because we already committed at that point, so it's no-op. In the meantime, the stealer would have put their slot in the state, which we would take here